OpenFeature/src/main/java/dev/openfeature/sdk/events/EventHandler.kt

```diff
@@ -12,21 +11,18 @@ import kotlinx.coroutines.flow.filterIsInstance
 import kotlinx.coroutines.launch
 import kotlin.reflect.KClass
 
-interface ProviderStatus {
-    fun isProviderReady(): Boolean
-}
-
-interface EventObserver {
+interface ProviderEventObserver {
     fun <T : OpenFeatureEvents> observe(kClass: KClass<T>): Flow<T>
+    fun isProviderReady(): Boolean
 }
 
 interface EventsPublisher {
     fun publish(event: OpenFeatureEvents)
 }
 
-inline fun <reified T : OpenFeatureEvents> EventObserver.observe() = observe(T::class)
+inline fun <reified T : OpenFeatureEvents> ProviderEventObserver.observe() = observe(T::class)
 
-class EventHandler(dispatcher: CoroutineDispatcher) : EventObserver, EventsPublisher, ProviderStatus {
+class EventHandler(dispatcher: CoroutineDispatcher) : ProviderEventObserver, EventsPublisher {
```

Comment:
the event handler right now (before this PR) is singleton for the whole app life cycle, after this PR, the event handler will be one instance inside each provider. making it provider level instead of app level.

about removing `EventPublisher` interface, we need to have a way to only expose the event observer for observing the events coming from the provider without exposing publishing events function. I believe we need this 2 interfaces in this case, wdyt?

Reply:
I guess what is confusing me is why we are writing an EventPublisher implementation on the SDK layer, since the SDK is not publishing any event. This makes me think that maybe each Provider should implement `ProviderEventObserver` in the way they prefer, and not depend on the `EventHandler` at all.

That being said, Providers today [do depend on EventHandler](https://github.com/spotify/confidence-openfeature-provider-kotlin/blob/421782a2b5c62b15a8b29e38e1ad29bac035b4a8/Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt#L256), so this PR is not a regression in this sense. Let's tackle this comment thread in a future PR if necessary

Reply:
valid question, I suppose that using the `EventHandler` is optional, they can still do their own eventing, but this is the way that sdk has some utility .. like we did the hard part for you now just use it .. basically some common utility that can be useful for other providers. maybe in the future this can be in another package, like kotlin-sdk-extensions .. then people can use that package if they need some other functionality .. also async stuff to be moved to there.